### PR TITLE
Remove functions autoexection

### PR DESCRIPTION
**Pull request by jccazeaux**
PR for #571 

Includes tests and docs.
My English might not be good so do not hesitate to be criticize the docs !
If anyone has a better idea to the backward compatibility parameter (I chose `bindingAutoexecuteFunctions`), let me know.

### CODE REVIEW

**Comment by Leeds-eBooks** [comment]
Thanks, this looks great to me. I would probably simplify the config to `executeFunctions`.

File: docs/docs/guide/_sections/functions-call.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+To call a function in an expression, rivets includes a special formatter `call`. This formatter will call the function, any formatter parameter will become an argument sent to the function. The following text binding will call the function `event.duration` with the two arguments `event;start` and `event.end`.
```
Comment (Leeds-eBooks) [inline docs/docs/guide/_sections/functions-call.md]: Sorry just 2 things: typo on `event;start`, and could you add the explanation "event.duration will be called again every time `event.start` and `event.end` change"?

File: spec/rivets/formatters.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    beforeEach(function() {
+      model = {
+        fn: function(arg1, arg2) {
+          return '' + arg1 + arg2;
+        }
+      }
+    })
+
+    it("calls function with arguments", function() {
+      rivets.formatters['call'](model.fn, 'foo', 'bar').should.equal('foobar')
```
Comment (Leeds-eBooks) [inline spec/rivets/formatters.js]: Do you think there should be a test using `this` inside the bound function too?

Comment (jccazeaux) [inline spec/rivets/formatters.js]: Yep
I'll commit it

File: docs/docs/guide/_sections/functions-call.md (hunk 1 above)
Comment (jccazeaux) [inline docs/docs/guide/_sections/functions-call.md]: Thanks, it's done

**Pull request merged**